packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts

```diff
@@ -959,4 +969,48 @@ export class ObjectMetadataService extends TypeOrmQueryService<ObjectMetadataEnt
 
     return { favoriteObjectMetadata };
   }
+
+  private async createMigrationToAlterCommentOnForeignKeyDeletion(
+    workspaceId: string,
+    relationToDelete: RelationToDelete,
+  ) {
+    const dataSourceMetadata =
+      await this.dataSourceService.getLastDataSourceMetadataFromWorkspaceIdOrFail(
+        workspaceId,
+      );
+
+    const workspaceDataSource =
+      await this.typeORMService.connectToDataSource(dataSourceMetadata);
+
+    const alteredComment = await buildAlteredCommentOnForeignKeyDeletion(
+      relationToDelete.toObjectName,
+      relationToDelete.fromObjectName,
+      dataSourceMetadata.schema,
+      workspaceDataSource,
+    );
+
+    if (alteredComment) {
+      await this.workspaceMigrationService.createCustomMigration(
+        generateMigrationName(
+          `alter-comment-${relationToDelete.fromObjectName}-${relationToDelete.toObjectName}`,
+        ),
+        workspaceId,
+        [
+          {
+            name: computeTableName(
+              relationToDelete.toObjectName,
+              relationToDelete.toObjectMetadataIsCustom,
+            ),
+            action: WorkspaceMigrationTableActionType.ALTER,
+            columns: [
+              {
+                action: WorkspaceMigrationColumnActionType.CREATE_COMMENT,
```

Comment:
is the comment on the column or on the table? It feels weird to be in WorkspaceMigrationColumnActionType

Reply:
this is on the table. I agree this is weird, but I do not like the current migration structure currently so I would first start by global refacto before touching it. I posted it some time ago in discord.

For example there, I would have to add a field nullable `comment` instead of adding a column action. And then, when I receive `WorkspaceMigrationTableActionType.ALTER`, check if comment is set.

That would be much cleaner to have `WorkspaceMigrationTableActionType` with more granular types, such as `OVERRIDE_COMMENT`